lib/msf/core/exploit/remote/http/http_cookie_jar.rb

```diff
@@ -29,8 +29,12 @@ def initialize
           def add(cookie)
             raise TypeError, "Passed cookie is of class '#{cookie.class}' and not a subclass of '#{Msf::Exploit::Remote::HTTP::HttpCookie}" unless cookie.is_a?(Msf::Exploit::Remote::HTTP::HttpCookie)
 
-            @cookie_jar.add(cookie)
-            self
+            begin
+              @cookie_jar.add(cookie)
+              self
+            rescue ArgumentError => e
+              elog("Cookie #{cookie.to_s} could not be added to jar. Error was thrown with following message:\n#{e.message}")
```

Comment:
I think `elog` can have the error passed in as an argument like this
```suggestion
              elog("Cookie #{cookie.to_s} could not be added to jar. Error was thrown with following message:\n#{e.message}", error: e)
```